src/apps/climsoft/services/obselement_service.py

```diff
@@ -0,0 +1,165 @@
+import logging
+from typing import List
+from sqlalchemy.orm.session import Session
+from opencdms.models.climsoft import v4_1_1_core as models
+from src.apps.climsoft.schemas import obselement_schema
+from fastapi.exceptions import HTTPException
+
+logger = logging.getLogger("ClimsoftObsElementService")
+logging.basicConfig(level=logging.INFO)
+
+
+class FailedCreatingObsElement(Exception):
+    pass
+
+
+class FailedGettingObsElement(Exception):
+    pass
+
+
+class FailedGettingObsElementList(Exception):
+    pass
+
+
+class FailedUpdatingObsElement(Exception):
+    pass
+
+
+class FailedDeletingObsElement(Exception):
+    pass
+
+
+class ObsElementDoesNotExist(Exception):
+    pass
+
+
+def create(db_session: Session, data: obselement_schema.CreateObsElement) -> obselement_schema.ObsElement:
+    try:
+        obs_element = models.Obselement(**data.dict())
+        db_session.add(obs_element)
+        db_session.commit()
+        return obselement_schema.ObsElement.from_orm(obs_element)
+    except Exception as e:
+        db_session.rollback()
+        logger.exception(e)
+        raise FailedCreatingObsElement("Failed creating obs_element.")
+
+
+def get(db_session: Session, element_id: str) -> obselement_schema.ObsElement:
+    try:
+        obs_element = db_session.query(models.Obselement).filter_by(elementId=element_id).first()
+
+        if not obs_element:
+            raise HTTPException(status_code=404, detail="ObsElement does not exist.")
+
+        return obselement_schema.ObsElement.from_orm(obs_element)
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.exception(e)
+        raise FailedGettingObsElement("Failed getting obs_element.")
+
+
+def query(
+        db_session: Session,
+        element_id: str = None,
+        element_name: str = None,
+        abbreviation: str = None,
+        description: str = None,
+        element_scale: float = None,
+        upper_limit: float = None,
+        lower_limit: str = None,
+        units: str = None,
+        element_type: str = None,
+        qc_total_required: int = None,
+        selected: bool = None,
+        limit: int = 25,
+        offset: int = 0
+) -> List[obselement_schema.ObsElement]:
+    """
+    This function builds a query based on the given parameter and returns `limit` numbers of `obselement` row skipping
```

Comment:
I have made some assumption here about, how to use these parameters when making queries. Please suggest how it can be improved.

Reply:
I'll raise this as a new issue and see if we can make this a broader discussion to get more input